fission-cli/library/Fission/CLI/Release.hs

```diff
@@ -0,0 +1,63 @@
+module Fission.CLI.Release (checkLatestRelease) where
+
+import           Data.Versions
+import qualified Data.Yaml                               as YAML
+import qualified System.Console.ANSI                    as ANSI
+
+import           Fission.Prelude
+
+import           Fission.Error.NotFound.Types
+
+import           Fission.CLI.Environment                as Env
+import qualified Fission.CLI.Meta                       as Meta
+import           Fission.CLI.Display.Text
+import qualified Fission.CLI.Display.Error       as CLI.Error
+
+import qualified Fission.Internal.UTF8                  as UTF8
+
+import           GitHub (github')
+import qualified GitHub
+import           GitHub.Data.Releases                   as Releases
+
+checkLatestRelease ::
+  ( MonadLogger      m
+  , MonadIO          m
+  , MonadTime        m
+  , MonadCleanup     m
+  , MonadEnvironment m
+  , m `Raises` YAML.ParseException
+  , m `Raises` NotFound FilePath
+  , Show (OpenUnion (Errors m))
+  )
+  => m ()
+checkLatestRelease = do
+  attempt Env.get >>= \case
+    Left  err -> CLI.Error.put err "Unable to parse config"
+    Right Env {updateChecked} -> do
+      now <- currentTime
+      let
+        nextUpdateTime = addUTCTime 86400 updateChecked
+      unless (now < nextUpdateTime) do
+        logDebug @Text "Checking for newer versions..."
+        possibleVersion <- liftIO $ github' GitHub.latestReleaseR "fission-suite" "fission"
+        case possibleVersion of
```

Comment:
_Totally fine as-is._

However, if you want to cut a small amount of boilerplate, you can switch to `>>=` here to "handle the result directly". You tend to get an `Either` back when talking to external services, because there's any number of things that can go wrong in that call, so this is a very common pattern.

```haskell
liftIO (github' GitHub.latestRelaseR "fission-suite" "fission") >>= \case
  Left _ ->
    return ()

  Right latestRelease -> do
    -- ...
```